src/main/groovy/io/seqera/wave/service/blob/impl/KubeTransferStrategy.groovy

```diff
@@ -65,4 +83,18 @@ class KubeTransferStrategy implements TransferStrategy {
                 .putUnencodedChars(info.creationTime.toString())
                 .hash()
     }
+
+    void cleanup(String podName) {
+        try {
+            if(k8sService.getPod(podName).status.phase == 'Succeeded') {
+                k8sService.deletePod(podName)
+            }else if(k8sService.getPod(podName).status.phase == 'Running'){
+                sleep(5000) // sometimes pods is not completed when finally runs so wait for 5 seconds
+                k8sService.deletePod(podName)
+            }
```

Comment:
This is a bit too optimist. Let's do: 

1. turn into a method `deletePodWhenReachStatus(podName, statusName, timeout)` 
2. use a while to check if the status is the one expected or reply 
3. move as a servie method into K8s service

Reply:
done